python_modules/libraries/dagster-k8s/dagster_k8s/pipes.py

```diff
@@ -424,6 +425,10 @@ def run(
         ) as pipes_session:
             namespace = namespace or _detect_current_namespace(self.kubeconfig_file) or "default"
             pod_name = get_pod_name(context.run_id, context.op.name)
+
+            if not base_pod_spec or len(base_pod_spec.get("containers", [])) == 1:
+                image = image or context.dagster_run.tags.get(DOCKER_IMAGE_TAG)
```

Comment:
I think you could also fetch this using the same logic the run launcher does, but the run launcher does also set this tag, so I suppose either works.

Reply:
Do you want to assert in some fashion that it is non-null after doing this?